src/Set.ts

```diff
@@ -353,6 +353,17 @@ export function remove<A>(E: Eq<A>): (a: A) => (set: Set<A>) => Set<A> {
   return a => set => filter((ax: A) => !E.equals(a, ax))(set)
 }
 
+/**
+ * Checks an element is a member of a set;
+ * If yes, removes the value from the set
+ * If no, inserts the value to the set
+ *
+ * @since 2.5.0
+ */
+export function toggle<A>(E: Eq<A>): (a: A) => (set: Set<A>) => Set<A> {
+  return a => set => (elem(E)(a, set) ? remove : insert)(E)(a)(set)
```

Comment:
Maybe we could memoize `elem`, `remove` and `insert` here

```ts
export function toggle<A>(E: Eq<A>): (a: A) => (set: Set<A>) => Set<A> {
  const elemE = elem(E)
  const removeE = remove(E)
  const insertE = insert(E)
  return a => set => (elemE(a, set) ? removeE : insertE)(a)(set)
}
```

Reply:
Thank you!

I followed your suggestion in 0ba409ee7dcbe91fbd5b9848362e9e9d7b8d3391, and also added a change on the document.
If this change looks good for you, I'll squash my commits and push it again after your approval.